app/controllers/assignments_controller.rb

```diff
@@ -293,9 +242,130 @@ def update_nil_dd_description_url(due_date_all)
+      @assignment_submission_due_dates = @due_date_all.select {|due_date| due_date.deadline_type_id == DeadlineHelper::DEALINE_TYPE_SUBMISSION }
+      @assignment_review_due_dates = @due_date_all.select {|due_date| due_date.deadline_type_id == DeadlineHelper::DEALINE_TYPE_REVIEW }
+    end
+    @assignment_form.assignment.staggered_deadline == true
+  end
+
+  def check_due_date_nameurl_notempty(dd)
+    @due_date_nameurl_notempty = due_date_nameurl_notempty?(dd)
+    @due_date_nameurl_notempty_checkbox = @due_date_nameurl_notempty
+    @metareview_allowed = meta_review_allowed?(dd)
+    @drop_topic_allowed = drop_topic_allowed?(dd)
+    @signup_allowed = signup_allowed?(dd)
+    @team_formation_allowed = team_formation_allowed?(dd)
+  end
+
+  def adjust_timezone_when_due_date_present(dd)
+    if dd.due_at.present?
+      dd.due_at = dd.due_at.to_s.in_time_zone(current_user.timezonepref)
+    end
+  end
+
+  def validate_due_date
+    @due_date_nameurl_notempty && @due_date_nameurl_notempty_checkbox &&
+      (@metareview_allowed || @drop_topic_allowed || @signup_allowed || @team_formation_allowed)
+  end
+
+  def check_assignment_questionnaires_usage
+    @assignment_questionnaires.each do |aq|
+      unless aq.used_in_round.nil?
+        @reviewvarycheck = 1
+        break
+      end
+    end
+  end
+
+  def handle_rubrics_not_assigned_case
+    if !empty_rubrics_list.empty? && request.original_fullpath == "/assignments/#{@assignment_form.assignment.id}/edit"
+      rubrics_needed = needed_rubrics(empty_rubrics_list)
+      flash.now[:error] = "You did not specify all the necessary rubrics. You need " + rubrics_needed +
+          " of assignment <b>#{@assignment_form.assignment.name}</b> before saving the assignment. You can assign rubrics <a id='go_to_tabs2' style='color: blue;'>here</a>."
+    end
+  end
+
+  def handle_assignment_directory_path_nonexist_case
+    if @assignment_form.assignment.directory_path.nil? || @assignment_form.assignment.directory_path.empty?
+      flash.now[:error] = "You did not specify your submission directory."
+    end
+  end
+
+  # helper methods for update
+  def assignment_form_key_nonexist_case_handler
+    @assignment = Assignment.find(params[:id])
+    @assignment.course_id = params[:course_id]
+
+    if @assignment.save
+      flash[:note] = 'The assignment was successfully saved.'
+      redirect_to list_tree_display_index_path
+    else
+      flash[:error] = "Failed to save the assignment: #{@assignment.errors.full_messages.join(' ')}"
+      redirect_to edit_assignment_path @assignment.id
+    end
+  end
+
+  def retrieve_assignment_form
+    @assignment_form = AssignmentForm.create_form_object(params[:id])
+    @assignment_form.assignment.instructor ||= current_user
+    params[:assignment_form][:assignment_questionnaire].reject! do |q|
+      q[:questionnaire_id].empty?
+    end
+  end
+
+  def handle_current_user_timezonepref_nil
```

Comment:
Method `AssignmentsController#handle_current_user_timezonepref_nil` is defined at both /code/app/controllers/assignments_controller.rb:250 and /code/app/controllers/assignments_controller.rb:352.

Reply:
Please change the second one in 'update' method to the name 'handle_current_user_timezonepref_nil_with_parent_timezone'